doc/tools/coqrst/coqdomain.py

```diff
@@ -827,22 +827,41 @@ def split_sentences(source):
         return re.split(r"(?<=(?<!\.)\.)\s+", source)
 
     @staticmethod
-    def parse_options(options):
+    def parse_options(node):
         """Parse options according to the description in CoqtopDirective."""
-        opt_undo = 'undo' in options
+
+        options = node['coqtop_options']
+
+        # Behavior options
         opt_reset = 'reset' in options
-        opt_all, opt_none = 'all' in options, 'none' in options
-        opt_input, opt_output = opt_all or 'in' in options, opt_all or 'out' in options
+        opt_fail = 'fail' in options
+        opt_restart = 'restart' in options
+        opt_abort = 'abort' in options
+        options = options - set(('reset', 'fail', 'restart', 'abort'))
```

Comment:
Sets are usually written as `{…}` instead of `set((…))`

Reply:
Sorry, I'm not an advanced Pythonista. I only copied what was already there.